app/controllers/relationships_controller.rb

```diff
@@ -0,0 +1,40 @@
+class RelationshipsController < ApplicationController
+  before_action :logged_in_user
+  before_action :load_user, only: :create
+  before_action :load_relationship, only: :destroy
+
+  def create
+    current_user.follow(@user)
+    respond_to do |format|
+      format.html{redirect_to @user}
+      format.js
+    end
+  end
+
+  def destroy
+    user = @relationship.followed
+    current_user.unfollow(user)
+    respond_to do |format|
+      format.html{redirect_to @user}
+      format.js
+    end
+  end
+
+  private
+  def load_user
+    @user = User.find_by id: params[:followed_id]
+    return if @user
+
+    flash[:danger] = t ".not_found_user"
+    redirect_to root_url
+  end
+
+  def load_relationship
+    @relationship = Relationship.find_by id: params[:id]
+    @user = @relationship.followed if @relationship
```

Comment:
chỗ này ko sai, nhưng mess nên cụ thể hơn, nên tách cái load relationship ra thành 1 method riêng, xử lý khi relationship nil với mess báo lỗi relationship not found  và dùng action filter để gọi